alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,456 @@
+				Required: true,
+			},
+			"cluster_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"emr_ver": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"charge_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"host_group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_group_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"host_group_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"charge_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"period": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"node_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"auto_renew": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"vswitch_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"gpu_driver": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"is_open_public_ip": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"security_group_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"net_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"option_software_list": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"high_availability_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"use_local_metadb": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  "false",
```

Comment:
> If the paramete is `bool`， its default should be `false` not `"false"`.
> In addition, I think other bool parameters also should set default because of its final value is one of the `false` and `true`

fixed